File: src/name/soy/dc/servlet/ws/MonitorSession.java
package name.soy.dc.servlet.ws;

import lombok.Data;
import name.soy.dc.device.IDevice;

import javax.websocket.Session;
import java.util.ArrayList;

@Data
public class MonitorSession {
    public final Session session;
    boolean authed = false;

    public IDevice device;

    public MonitorSession(Session session) {
        this.session = session;
        if(!Monitor.dsessions.containsKey(device)){
            Monitor.dsessions.put(device, new ArrayList<>());
        }
        Monitor.dsessions.get(device).add(this);
    }

    public void disconnect(){

    }
}

File: src/name/soy/dc/servlet/ws/MonitorSession.kt
package name.soy.dc.servlet.ws

import name.soy.dc.device.IDevice
import javax.websocket.Session

class MonitorSession(val session: Session) {
	var authed = false
	var device: IDevice? = null
	fun disconnect() {}

	init {
		if (!Monitor.dsessions.containsKey(device)) {
			Monitor.dsessions[device] = arrayListOf<MonitorSession>()
		}
		Monitor.dsessions[device]!!.add(this)
	}
}